scrabble.js

```diff
@@ -1,14 +1,150 @@
+function Exception(message) {
+  this.message = message;
+  this.name = 'Exception';
+}
+
 const Scrabble = {
-  score: function(word) {
-    // TODO: implement score
-  }
+  scoreShart:  {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10
+  },
+  validWord(word) {
+    word = word.toUpperCase();
+    for (let letter of word) {
+      if (!Scrabble.scoreShart.hasOwnProperty(letter)) {
+        throw new Exception('InvalidInput');
+      }
+    }
+    return true;
+  },
+
+  score(word) {
+    let score = 0;
+    if (Scrabble.validWord(word)){
+      word = word.toUpperCase();
+      for (let letter of word) {
+        score += Scrabble.scoreShart[letter];
+      }
+    }
+
+    if (word.length ===7){
+      score+= 50;
+    } else if (word.length > 7) {
+      throw new Exception('The size of the word should be at most 7');
+    } else if (word.length === 0 ){
+      throw new Exception('The word should have length bigger than 0');
+    }
+    return score;
+  }, //close the score function
+
 
-  // TODO: add the highestScoreFrom method
+  highestScoreFrom(arrayOfWords) {
 
+    let all_scores = [];
+    for(let word of arrayOfWords) {
+      all_scores.push(Scrabble.score(word));
+    };
+    let maximo = Math.max.apply(null, all_scores);
+    let i = 0;
+    let maximosWords = [];
+    while (i < all_scores.length) {
```

Comment:
This algorithm ends up being a little over-complicated. By my reading, your steps are:

1. Calculate a score for every word
1. Find those words that share the highest score
1. Use apply tie-breaking logic to those words

A simpler algorithm is to iterate through the list of words, keeping track of the best word / score seen so far. For each word:
- If the current word has a higher score than the best word, it's the new best word
- If the best word has a higher score, keep it
- If the two are tied, apply tie-breaking logic, favoring the existing best word.

This single-pass approach is more concise and (in my opinion) easier to understand.